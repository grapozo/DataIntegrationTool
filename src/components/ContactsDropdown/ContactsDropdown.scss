@use './configuration/DropdownConfigurations' as *;
.contactsDropdown {
  position: relative;
  background-color: #f5f8fa;
  border: 1px solid #cbd6e2;
  border-radius: 0.125rem;
  height: $initial-dropdown-height;
  font-size: 1rem;
  user-select: none;
  transition: all $animation-speed;
  // Disables highlight when clicking the element (Chrome).
  -webkit-tap-highlight-color: transparent;

  .contactList {
    opacity: 0;
    visibility: hidden;
    // Animation for hiding/showing the component
    transition: visibility $animation-speed linear, opacity $animation-speed linear;
  }

  &.expanded {
    height: $expanded-dropdown-height;
  }
}
