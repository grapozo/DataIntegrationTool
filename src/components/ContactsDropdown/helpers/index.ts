export {
  getUpdatedSelectedContactsLabel,
  generateListOfContacts
} from './ContactsDropdown.helpers';
