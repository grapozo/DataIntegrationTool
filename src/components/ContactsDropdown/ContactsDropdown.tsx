import styles from './ContactsDropdown.scss';
import { useEffect, useState } from 'react';
import { Contact } from 'shared/interfaces/contact.interface';
import { ContactsDropdownList } from './ContactsDropdownList';
import { ContactsDropdownPlaceholder } from './ContactsDropdownPlaceholder';
import { getUpdatedSelectedContactsLabel } from './helpers';

type ContactsDropdownProps = {
  /** The contacts list */
  contacts: Contact[];
};

export const ContactsDropdown = ({ contacts }: ContactsDropdownProps): JSX.Element => {
  const [dropdownExpanded, setDropdownExpanded] = useState<boolean>(false);
  const [selectedContacts, setSelectedContacts] = useState<Contact[]>([]);
  // Since we start with no selected contacts, we default the label to 'No Contacts'.
  const [selectedContactsLabel, setSelectedContactsLabel] = useState<string>('No Contacts');

  const handleContactListChangedCallback = (selectedContacts: Contact[]) => {
    setSelectedContacts(selectedContacts);
  };

  /**
   * Runs when the selectedContacts array has been updated.
   */
  useEffect(() => {
    const updatedContactsLabel = getUpdatedSelectedContactsLabel(
      contacts.length,
      selectedContacts.length
    );
    setSelectedContactsLabel(updatedContactsLabel);
  }, [selectedContacts]);

  return (
    <div className={`${styles.contactsDropdown} ${dropdownExpanded ? styles.expanded : ''}`}>
      <ContactsDropdownPlaceholder
        onClick={() => setDropdownExpanded(!dropdownExpanded)}
        selectedContactsLabel={selectedContactsLabel}
        isDropdownExpanded={dropdownExpanded}
      />
      <ContactsDropdownList
        isVisible={dropdownExpanded}
        contacts={contacts}
        selectedContactListChangedCallback={handleContactListChangedCallback}
      />
    </div>
  );
};
