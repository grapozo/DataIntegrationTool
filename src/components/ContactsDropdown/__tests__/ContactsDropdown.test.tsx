import { create } from 'react-test-renderer';
import { ContactsDropdown } from '../ContactsDropdown';
import '@testing-library/jest-dom';

/**
 * For now, we only keep the snapshot test, since we already have
 * the tests for the most specific functionality of the component
 * down the tree.
 */
describe('ContactsDropdown', () => {
  it('should correctly render', () => {
    // Since this is a snap test, we keep a static list.
    const tree = create(
      <ContactsDropdown
        contacts={[
          {
            id: 'ef1934cc85aec5ef909f74c7778529ea',
            name: 'Mocked Contact'
          }
        ]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
