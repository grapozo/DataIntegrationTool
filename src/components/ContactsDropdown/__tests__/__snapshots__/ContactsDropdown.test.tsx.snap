// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsDropdown should correctly render 1`] = `
<div
  className="undefined "
>
  <div
    className="
        undefined 
        
      "
    onClick={[Function]}
  >
    <span
      className="undefined"
    >
      No Contacts
    </span>
  </div>
  <div
    className="
      undefined 
      "
  >
    <div>
      <div
        className="undefined"
      >
        <label>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          Mocked Contact
        </label>
      </div>
    </div>
  </div>
</div>
`;
